internal/kubernetes/operator/secrets/secrets.go

```diff
@@ -21,27 +21,67 @@ import (
 )
 
 const (
-	TypeLabelKey      = "atlas.mongodb.com/type"
-	CredLabelVal      = "credentials"
-	PasswordField     = "password"
-	CredPrivateAPIKey = "privateApiKey"
-	CredPublicAPIKey  = "publicApiKey"
-	CredOrgID         = "orgId"
+	TypeLabelKey        = "atlas.mongodb.com/type"
+	ProjectIDLabelKey   = "atlas.mongodb.com/project-id"
+	ProjectNameLabelKey = "atlas.mongodb.com/project-name"
+	NotifierIDLabelKey  = "atlas.mongodb.com/notifier-id"
+	CredLabelVal        = "credentials"
+	PasswordField       = "password"
+	CredPrivateAPIKey   = "privateApiKey"
+	CredPublicAPIKey    = "publicApiKey"
+	CredOrgID           = "orgId"
 )
 
-func NewAtlasSecret(name, namespace string, data map[string][]byte, dictionary map[string]string) *corev1.Secret {
-	return &corev1.Secret{
-		TypeMeta: v1.TypeMeta{
-			Kind:       "Secret",
-			APIVersion: "v1",
-		},
-		ObjectMeta: v1.ObjectMeta{
-			Name:      resources.NormalizeAtlasName(name, dictionary),
-			Namespace: namespace,
-			Labels: map[string]string{
-				TypeLabelKey: CredLabelVal,
+type AtlasSecretBuilder func() (*corev1.Secret, map[string]string)
+
+func NewAtlasSecretBuilder(name, namespace string, dictionary map[string]string) AtlasSecretBuilder {
```

Comment:
Interesting approach :) This can also be implemented with a struct, eliminating the `Build()` method at the end.

Reply:
Indeed, it was something real quick, I don't think we need anything fancy here. If we're going to need more initialising variables we can also introduce a struct. The `Build` method is just to denote that this is a builder so its implementation can vary independently.